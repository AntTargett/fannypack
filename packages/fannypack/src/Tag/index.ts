import Tag from './Tag';
import * as styledTag from './styled';

export default Tag;

export * from './Tag';
export { default as Tag } from './Tag';

export { styledTag };
