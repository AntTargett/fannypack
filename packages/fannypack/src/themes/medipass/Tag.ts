import { css, space } from '../../styled';

export default {
  base: css`
    border-radius: 30px;
    text-transform: uppercase;
  `,
  TagContent: {
    base: css`
      padding: 0 ${space(3)}em;
    `
  }
};
