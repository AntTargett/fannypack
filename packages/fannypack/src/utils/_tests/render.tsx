import * as React from 'react';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (Component: any, { theme = {} } = {}) => render(<Provider theme={theme}>{Component}</Provider>);
