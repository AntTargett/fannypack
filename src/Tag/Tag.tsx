import * as React from 'react';
import * as PropTypes from 'prop-types';
import { InlineBlockProps as ReakitInlineBlockProps } from '@jmoxey/reakit/ts/InlineBlock/InlineBlock';

import { Omit } from '../types';
import _Tag from './styled';

export type LocalTagProps = {
  className?: string;
  children: React.ReactNode;
  kind?: 'outlined';
  palette?: string;
  size?: string;
};
export type TagProps = Omit<ReakitInlineBlockProps, 'size'> & LocalTagProps;

export const Tag: React.FunctionComponent<LocalTagProps> = ({ children, size, ...props }) => (
  <_Tag styledSize={size} {...props}>
    {children}
  </_Tag>
);

Tag.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
  kind: PropTypes.oneOf(['outlined']) as PropTypes.Validator<LocalTagProps['kind']>,
  palette: PropTypes.string,
  size: PropTypes.string
};
Tag.defaultProps = {
  className: undefined,
  kind: undefined,
  palette: 'text',
  size: undefined
};

// @ts-ignore
const C: React.FunctionComponent<TagProps> = Tag;
export default C;
