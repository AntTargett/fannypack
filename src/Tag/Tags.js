// @flow
import React, { type Node } from 'react';

import { Tags as _Tags } from './styled';

type Props = {
  children: Node,
  className?: string
};

const Tags = ({ children, className, ...props }: Props) => (
  <_Tags className={className} {...props}>
    {children}
  </_Tags>
);

Tags.defaultProps = {
  className: null
};

export default Tags;
