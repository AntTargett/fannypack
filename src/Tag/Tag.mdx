---
name: Tag
route: /components/tag
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Tag from '../Tag';

# Tag

## Import

`import { Tag } from 'fannypack'`

## Basic Usage

<Playground>
  <Tag>Default</Tag>
  <Tag textTransform="uppercase" marginLeft="xxxsmall">Uppercase</Tag>
</Playground>

## Colors

<Playground>
  <Tag palette="primary">Primary</Tag>
  <Tag palette="secondary" marginLeft="xxxsmall">Secondary</Tag>
  <Tag palette="success" marginLeft="xxxsmall">Success</Tag>
  <Tag palette="warning" marginLeft="xxxsmall">Warning</Tag>
  <Tag palette="danger" marginLeft="xxxsmall">Danger</Tag>
</Playground>

### Tints

<Playground>
  <Tag palette="textTint">Default</Tag>
  <Tag palette="primaryTint" marginLeft="xxxsmall">Primary</Tag>
  <Tag palette="secondaryTint" marginLeft="xxxsmall">Secondary</Tag>
  <Tag palette="successTint" marginLeft="xxxsmall">Success</Tag>
  <Tag palette="warningTint" marginLeft="xxxsmall">Warning</Tag>
  <Tag palette="dangerTint" marginLeft="xxxsmall">Danger</Tag>
</Playground>

## Sizes

<Playground>
  <Tag>Default</Tag>
  <Tag size="medium" marginLeft="xxxsmall">Medium</Tag>
  <Tag size="large" marginLeft="xxxsmall">Large</Tag>
</Playground>

## Props

<PropsTable of={Tag} />

## Theming

### Schema

```jsx
{
  base: string | Object,
  sizes: {
    medium: string | Object,
    large: string | Object
  }
}
```
