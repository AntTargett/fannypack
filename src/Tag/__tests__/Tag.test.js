import React from 'react';
import render from '../../_utils/tests/render';
import Tag from '../Tag';
import 'jest-styled-components';

it('renders correctly', () => {
  const { container } = render(<Tag>Tag</Tag>);
  expect(container.firstChild).toMatchSnapshot();
});

describe('colors', () => {
  ['danger', 'success', 'warning', 'primary'].forEach(color => {
    it(`renders correctly for a tag with color ${color}`, () => {
      const { container } = render(<Tag palette={color} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('sizes', () => {
  ['medium', 'large'].forEach(size => {
    it(`renders correctly for a tag with size ${size}`, () => {
      const { container } = render(<Tag size={size} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
