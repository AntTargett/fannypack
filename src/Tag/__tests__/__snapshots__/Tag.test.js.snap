// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colors renders correctly for a tag with color danger 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c4 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #d60027;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c3 {
  display: inline-block;
}

.c2 {
  background-color: #d60027;
  border-radius: 4px;
  color: #fff;
  font-size: 0.8rem;
  font-weight: 500;
  padding: 0.2em 0.4em;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3 c4"
  />
</div>
`;

exports[`colors renders correctly for a tag with color primary 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c4 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #3926a5;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c3 {
  display: inline-block;
}

.c2 {
  background-color: #3926a5;
  border-radius: 4px;
  color: #fff;
  font-size: 0.8rem;
  font-weight: 500;
  padding: 0.2em 0.4em;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3 c4"
  />
</div>
`;

exports[`colors renders correctly for a tag with color success 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c4 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #007b2e;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c3 {
  display: inline-block;
}

.c2 {
  background-color: #007b2e;
  border-radius: 4px;
  color: #fff;
  font-size: 0.8rem;
  font-weight: 500;
  padding: 0.2em 0.4em;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3 c4"
  />
</div>
`;

exports[`colors renders correctly for a tag with color warning 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c4 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #ffb300;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c3 {
  display: inline-block;
}

.c2 {
  background-color: #ffb300;
  border-radius: 4px;
  color: #000;
  font-size: 0.8rem;
  font-weight: 500;
  padding: 0.2em 0.4em;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3 c4"
  />
</div>
`;

exports[`renders correctly 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c4 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #435a6f;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c3 {
  display: inline-block;
}

.c2 {
  background-color: #435a6f;
  border-radius: 4px;
  color: #fff;
  font-size: 0.8rem;
  font-weight: 500;
  padding: 0.2em 0.4em;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3 c4"
  >
    Tag
  </div>
</div>
`;

exports[`sizes renders correctly for a tag with size large 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c4 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #435a6f;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c3 {
  display: inline-block;
}

.c2 {
  background-color: #435a6f;
  border-radius: 4px;
  color: #fff;
  font-size: 0.8rem;
  font-weight: 500;
  padding: 0.2em 0.4em;
}

.c2 {
  font-size: 1.25em;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3 c4"
    style="size: large;"
  />
</div>
`;

exports[`sizes renders correctly for a tag with size medium 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c4 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #435a6f;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c3 {
  display: inline-block;
}

.c2 {
  background-color: #435a6f;
  border-radius: 4px;
  color: #fff;
  font-size: 0.8rem;
  font-weight: 500;
  padding: 0.2em 0.4em;
}

.c2 {
  font-size: 1em;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3 c4"
    style="size: medium;"
  />
</div>
`;
